File: srccode/frontend/src/services/userService.js
import API from './api';

export const getAllUsers = async () => {
  const response = await API.get('/users');
  return response.data;
};

export const getStaffCustomers = async () => {
  const response = await API.get('/users/customers');
  return response.data;
};

export const updateUserRole = async (userId, roleName, currentUserId) => {
  const response = await API.put(`/users/${userId}/role`, {
    roleName,
    currentUserId
  });

  return response.data;
};

export const updateUserStatus = async (userId, isActive, currentUserId) => {
  const response = await API.put(`/users/${userId}/status`, {
    isActive,
    currentUserId
  });

  return response.data;
};

File: srccode/backend/restaurant_management_system/src/main/java/com/rms/restaurant_management_system/service/interfaces/UserService.java
package com.rms.restaurant_management_system.service.interfaces;

import com.rms.restaurant_management_system.dto.request.UpdateUserRoleRequest;
import com.rms.restaurant_management_system.dto.request.UpdateUserStatusRequest;
import com.rms.restaurant_management_system.dto.response.StaffCustomerResponse;
import com.rms.restaurant_management_system.dto.response.UserResponse;

import java.util.List;

public interface UserService {

    UserResponse getProfileByEmail(String email);

    List<UserResponse> getAllUsers();

    List<StaffCustomerResponse> getStaffCustomers();

    UserResponse updateUserRole(Long userId, UpdateUserRoleRequest request);

    UserResponse updateUserStatus(Long userId, UpdateUserStatusRequest request);
}